apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/responses/components/ResponseCardModal.tsx

```diff
@@ -0,0 +1,105 @@
+import { ChevronLeft, ChevronRight, XIcon } from "lucide-react";
+import { useEffect, useState } from "react";
+import { TEnvironment } from "@formbricks/types/environment";
+import { TResponse } from "@formbricks/types/responses";
+import { TSurvey } from "@formbricks/types/surveys/types";
+import { TTag } from "@formbricks/types/tags";
+import { TUser } from "@formbricks/types/user";
+import { Button } from "@formbricks/ui/Button";
+import { SingleResponseCard } from "@formbricks/ui/SingleResponseCard";
+
+interface ResponseCardModalProps {
+  responses: TResponse[];
+  selectedResponse: TResponse | null;
+  setSelectedResponse: (response: TResponse | null) => void;
+  survey: TSurvey;
+  environment: TEnvironment;
+  user?: TUser;
+  environmentTags: TTag[];
+  updateResponse: (responseId: string, updatedResponse: TResponse) => void;
+  deleteResponses: (responseIds: string[]) => void;
+  isViewer: boolean;
+}
+
+export const ResponseCardModal = ({
+  responses,
+  selectedResponse,
+  setSelectedResponse,
+  survey,
+  environment,
+  user,
+  environmentTags,
+  updateResponse,
+  deleteResponses,
+  isViewer,
+}: ResponseCardModalProps) => {
+  const [currentIndex, setCurrentIndex] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (selectedResponse) {
+      const index = responses.findIndex((response) => response.id === selectedResponse.id);
+      setCurrentIndex(index);
+    }
+  }, [selectedResponse, responses]);
+
+  const handleNext = () => {
+    if (currentIndex !== null && currentIndex < responses.length - 1) {
+      setSelectedResponse(responses[currentIndex + 1]);
+    }
+  };
+
+  const handleBack = () => {
+    if (currentIndex !== null && currentIndex > 0) {
+      setSelectedResponse(responses[currentIndex - 1]);
+    }
+  };
+
+  const handleClose = () => {
+    setSelectedResponse(null);
+  };
+
+  // If no response is selected or currentIndex is null, do not render the modal
+  if (selectedResponse === null || currentIndex === null) return null;
+
+  return (
+    <div className="fixed inset-0 z-50 flex items-center justify-center bg-black bg-opacity-50 backdrop-blur-md">
```

Comment:
this file is actually rendering a `div` instead of the `Modal` component that we already have, which is why this container is not closable on pressing the escape key for example and has no opening animation, pls change this so that it actually renders a `Modal` component (from `packages/ui/Modal`) and can be closable on `ESC` key and have smooth animations while opening / closing 😊

Reply:
The reason behind not using the Modal component is this UI below 
<img width="547" alt="Screenshot 2024-09-02 at 4 02 10 PM" src="https://github.com/user-attachments/assets/492ae673-2f31-4b19-a169-ebf729c87928">
We wanted buttons to toggle between responses on the side of X button in modal, but that was not possible with the modal that we had. So used a custom one here